pkg/tenant/auth.go

```diff
@@ -13,6 +14,115 @@ import (
 // Set the maximum age of login protection cookies.
 const doubleCookiesMaxAge = time.Minute * 10
 
+// Register is a publically accessible endpoint that allows new users to create an
+// account via Quarterdeck by providing an email address and password.
+//
+// Route: POST /v1/register
+func (s *Server) Register(c *gin.Context) {
+	var err error
+
+	// Parse the request body
+	params := &api.RegisterRequest{}
+	if err = c.BindJSON(params); err != nil {
+		log.Warn().Err(err).Msg("could not parse request body")
+		c.JSON(http.StatusBadRequest, api.ErrorResponse("could not parse register request"))
+		return
+	}
+
+	// Validate that required fields were provided
+	if params.Name == "" || params.Email == "" || params.Password == "" || params.PwCheck == "" {
+		c.JSON(http.StatusBadRequest, api.ErrorResponse("missing required fields for registration"))
+		return
+	}
+
+	// Simple validation of the provided password
+	// Note: Quarterdeck also checks this along with password strength, but this allows
+	// us to filter some bad requests before they reach Quarterdeck.
+	if params.Password != params.PwCheck {
+		c.JSON(http.StatusBadRequest, api.ErrorResponse("passwords do not match"))
+		return
+	}
+
+	// Make the register request to Quarterdeck
+	req := &qd.RegisterRequest{
+		Name:     params.Name,
+		Email:    params.Email,
+		Password: params.Password,
+		PwCheck:  params.PwCheck,
+	}
+
+	// TODO: Handle error status codes returned by Quarterdeck
+	var reply *qd.RegisterReply
+	if reply, err = s.quarterdeck.Register(c.Request.Context(), req); err != nil {
+		log.Error().Err(err).Msg("could not register user")
+		c.JSON(http.StatusInternalServerError, api.ErrorResponse("could not complete registration"))
+		return
+	}
+
+	// TODO: Send verification email to the provided email address
```

Comment:
We'll have to discuss who's responsibility this is - Quarterdeck or Tenant?

Reply:
Hmm good question